app/content-panel/blogs/[id]/page.tsx

```diff
@@ -0,0 +1,129 @@
+'use client';
+
+import React, { useState, useEffect } from 'react';
+import styles from './BlogDetailsPage.module.scss';
+import BlogDetailsCard from '../../../../components/reusable-components/blogDetails-card/BlogDetailsCard';
+import useGetBlogDetails from '../../../../api/queries/blogs/getBlogDetails';
+
+const BlogDetailsPage = ({ params }: { params: { id: string } }) => {
+  const [isEditable, setIsEditable] = useState(false);
+  const [selectedImage, setSelectedImage] = useState<string | null>(null);
+  const [blogDetailsData, setBlogDetailsData] = useState({
+    title: '',
+    image: '',
+    content: '',
+    author: {
+      first_name: '',
+      last_name: '',
+    },
+    publishDate: '',
+    tags: [] as string[],
+  });
```

Comment:
You should create an interface for the BlogDetailsData and use it to infer types in the useState